File: code/zato-web-admin/src/zato/admin/static/js/channel/hl7/rest.js

// ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

$.fn.zato.data_table.HL7RESTChannel = new Class({
    toString: function() {
        var s = '<HL7RESTChannel id:{0} name:{1} is_active:{2}>';
        return String.format(s, this.id ? this.id : '(none)',
                                this.name ? this.name : '(none)',
                                this.is_active ? this.is_active : '(none)');
    }
});

// ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

$(document).ready(function() {
    $('#data-table').tablesorter();
    $.fn.zato.data_table.class_ = $.fn.zato.data_table.HL7RESTChannel;
    $.fn.zato.data_table.new_row_func = $.fn.zato.channel.hl7.rest.data_table.new_row;
    $.fn.zato.data_table.parse();
    $.fn.zato.data_table.setup_forms(['name', 'url_path', 'service', 'security_id', 'hl7_version']);
})

// ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

$.fn.zato.channel.hl7.rest.create = function() {
    $.fn.zato.data_table._create_edit('create', 'Create a new HL7 REST channel', null);
}

// ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

$.fn.zato.channel.hl7.rest.edit = function(id) {
    $.fn.zato.data_table._create_edit('edit', 'Update the HL7 REST channel', id);
}

// ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

$.fn.zato.channel.hl7.rest.data_table.new_row = function(item, data, include_tr) {
    let row = '';

    if(include_tr) {
        row += String.format("<tr id='tr_{0}' class='updated'>", item.id);
    }

    let is_active = item.is_active == true;
    let cluster_id = $(document).getUrlParam('cluster');

    row += "<td class='numbering'>&nbsp;</td>";
    row += "<td class='impexp'><input type='checkbox' /></td>";

    // 1
    row += String.format('<td>{0}</td>', item.name);
    row += String.format('<td>{0}</td>', is_active ? 'Yes' : 'No');
    row += String.format('<td>{0}</td>', item.hl7_version);

    // 2
    row += String.format('<td>{0}</td>', item.url_path);
    row += String.format('<td>{0}</td>', $.fn.zato.data_table.service_text(item.service, cluster_id));
    row += String.format('<td>{0}</td>', data.sec_def_link || $.fn.zato.empty_value);

    // 3
    row += String.format('<td>{0}</td>', String.format("<a href=\"javascript:$.fn.zato.channel.hl7.rest.edit('{0}')\">Edit</a>", item.id));
    row += String.format('<td>{0}</td>', String.format("<a href='javascript:$.fn.zato.channel.hl7.rest.delete_({0});'>Delete</a>", item.id));
    row += String.format("<td class='ignore item_id_{0}'>{0}</td>", item.id);

    // 4
    row += String.format("<td class='ignore'>{0}</td>", item.is_active);
    row += String.format("<td class='ignore'>{0}</td>", item.service);
    row += String.format("<td class='ignore'>{0}</td>", item.security);

    // 5
    row += String.format("<td class='ignore'>{0}</td>", item.json_path);
    row += String.format("<td class='ignore'>{0}</td>", item.should_parse_on_input);
    row += String.format("<td class='ignore'>{0}</td>", item.should_validate);

    // 6
    row += String.format("<td class='ignore'>{0}</td>", item.data_encoding);
    row += String.format("<td class='ignore'>{0}</td>", item.should_return_errors);

    if(include_tr) {
        row += '</tr>';
    }

    return row;
}

// ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

$.fn.zato.channel.hl7.rest.delete_ = function(id) {
    $.fn.zato.data_table.delete_(id, 'td.item_id_',
        'HL7 REST channel `{0}` deleted',
        'Are you sure you want to delete HL7 REST channel `{0}`?',
        true);
}

// ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

File: code/zato-web-admin/src/zato/admin/web/views/channel/hl7/rest.py
# -*- coding: utf-8 -*-

"""
Copyright (C) Zato Source s.r.o. https://zato.io

Licensed under LGPLv3, see LICENSE.txt for terms and conditions.
"""

from __future__ import absolute_import, division, print_function, unicode_literals

# Zato
from zato.admin.web.forms.channel.hl7.rest import CreateForm, EditForm
from zato.admin.web.views import CreateEdit, Delete as _Delete, django_url_reverse, extract_security_id, get_outconn_rest_list, \
     id_only_service, Index as _Index
from zato.common.api import CONNECTION, DATA_FORMAT, HL7, SEC_DEF_TYPE, SEC_DEF_TYPE_NAME, URL_TYPE, ZATO_NONE
from zato.common.json_internal import dumps
from zato.common.model import HL7Channel

# ################################################################################################################################
# ################################################################################################################################

class Index(_Index):
    method_allowed = 'GET'
    url_name = 'channel-hl7-rest'
    template = 'zato/channel/hl7/rest.html'
    service_name = 'zato.http-soap.get-list'
    output_class = HL7Channel
    paginate = True

    def get_initial_input(self):

        return {
            'connection': CONNECTION.CHANNEL,
            'transport': URL_TYPE.PLAIN_HTTP,
            'data_format': DATA_FORMAT.HL7
        }

    class SimpleIO(_Index.SimpleIO):
        input_required = 'cluster_id',
        output_required = 'id', 'name', 'is_active', 'is_internal', 'hl7_version', 'url_path', 'service_name', 'security_name', \
            'security_id', 'sec_type', 'sec_type_name', 'data_format',
        output_optional = 'json_path', 'should_parse_on_input', 'should_validate', 'should_return_errors'
        output_repeated = True

# ################################################################################################################################

    def on_before_append_item(self, item):
        if item.security_id and item.security_id != ZATO_NONE:
            item.sec_type_name = SEC_DEF_TYPE_NAME[item.sec_type]

        return item

# ################################################################################################################################

    def handle(self):
        security_list = self.get_sec_def_list(SEC_DEF_TYPE.BASIC_AUTH)
        return {
            'create_form': CreateForm(security_list, req=self.req),
            'edit_form': EditForm(security_list, prefix='edit', req=self.req),
        }

# ################################################################################################################################
# ################################################################################################################################

class _CreateEdit(CreateEdit):
    method_allowed = 'POST'

    class SimpleIO(CreateEdit.SimpleIO):
        input_required = 'name', 'is_internal', 'hl7_version', 'url_path', 'service', 'security_id', 'data_format'
        input_optional = 'is_active', 'json_path', 'should_parse_on_input', 'should_validate', 'should_return_errors'
        output_required = 'id', 'name'

# ################################################################################################################################

    def populate_initial_input_dict(self, initial_input_dict):
        initial_input_dict['connection'] = CONNECTION.CHANNEL
        initial_input_dict['transport'] = URL_TYPE.PLAIN_HTTP
        initial_input_dict['data_format'] = HL7.Const.Version.v2.id
        initial_input_dict['data_encoding'] = 'utf-8'

# ################################################################################################################################

    def pre_process_input_dict(self, input_dict):
        input_dict['security_id'] = extract_security_id(input_dict)

# ################################################################################################################################

    def post_process_return_data(self, return_data):
        return_data['sec_def_link'] = self.build_sec_def_link_by_input(self.input)

# ################################################################################################################################

    def success_message(self, item):
        return 'Successfully {} HL7 REST channel `{}`'.format(self.verb, item.name)

# ################################################################################################################################
# ################################################################################################################################

class Create(_CreateEdit):
    url_name = 'channel-hl7-rest-create'
    service_name = 'zato.http-soap.create'

# ################################################################################################################################
# ################################################################################################################################

class Edit(_CreateEdit):
    url_name = 'channel-hl7-rest-edit'
    form_prefix = 'edit-'
    service_name = 'zato.http-soap.edit'

# ################################################################################################################################
# ################################################################################################################################

class Delete(_Delete):
    url_name = 'channel-hl7-rest-delete'
    error_message = 'Could not delete HL7 REST channel'
    service_name = 'zato.http-soap.delete'

# ################################################################################################################################
# ################################################################################################################################
